src/main/java/org/fcrepo/storage/ocfl/DefaultOcflObjectSession.java

```diff
@@ -487,13 +503,16 @@ public synchronized void rollback() {
 
             LOG.info("Rolling back {} version {}", ocflObjectId, newVersionNum);
 
-            if (VersionNum.V1.equals(newVersionNum) ||
-                    (VersionNum.fromInt(2).equals(newVersionNum) && ocflRepo.hasStagedChanges(ocflObjectId))) {
+            if (isNewObject) {
                 // Purge the object if it only has one version or if it is a newly created object with a mutable head
```

Comment:
Does this comment still apply?  In other words does isNewObject = true man  "it only has one version or if it is a newly created object with a mutable head"

Reply:
Yes, that is still true. The original code was describing the state of the object when it is first created. If auto-versioning is used, the first version is 1. If auto-versioning is not used, the first version is 2, because an empty v1 must be created before creating the mutable head.

Reply:
Okay - good to know.